build.gradle

```diff
@@ -11,41 +11,90 @@ buildscript {
 }
 
 apply plugin: 'kotlin'
-apply plugin: 'maven-publish' // for local publishing in dev, not for pushing to Nexus
+apply plugin: 'maven'
+apply plugin: 'signing'
 
 group group_id
 version "$base_version$version_suffix"
+archivesBaseName = "kestrel"
 
 sourceSets {
     main.kotlin.srcDirs = main.java.srcDirs = ["src/main/kotlin"]
     test.kotlin.srcDirs = test.java.srcDirs = ["src/test/kotlin"]
 }
 
-java {
-    withJavadocJar()
-    withSourcesJar()
+
+task javadocJar(type: Jar) {
+    classifier = 'javadoc'
+    from javadoc
 }
 
-publishing {
-    publications {
-        library(MavenPublication) {
-            from components.java
-        }
-    }
+task sourcesJar(type: Jar) {
+    classifier = 'sources'
+    from sourceSets.main.allSource
+}
 
+artifacts {
+    archives javadocJar, sourcesJar
+}
+
+uploadArchives {
     repositories {
-        maven {
-            name 'nexus'
-            def isSnapshot = version.endsWith('SNAPSHOT')
-            url "$nexus_repository_root${isSnapshot ? 'snapshots' : 'releases'}/"
-            credentials {
-                username = System.getenv('NEXUS_USERNAME')
-                password = System.getenv('NEXUS_PASSWORD')
+        mavenDeployer {
+            beforeDeployment { MavenDeployment deployment -> signing.signPom(deployment) }
+
+            repository(url: "https://oss.sonatype.org/service/local/staging/deploy/maven2/") {
+                authentication(userName: System.getenv('SONATYPE_USERNAME'), password: System.getenv('SONATYPE_PASSWORD'))
+            }
+
+            snapshotRepository(url: "https://oss.sonatype.org/content/repositories/snapshots/") {
+                authentication(userName: System.getenv('SONATYPE_USERNAME'), password: System.getenv('SONATYPE_PASSWORD'))
+            }
+
+            pom.project {
+                name 'Kestrel'
+                packaging 'jar'
+                // optionally artifactId can be defined here
+                description 'Kotlin Framework for running event-sourced services'
+                url 'https://github.com/cultureamp/kestrel'
+
+                scm {
+                    connection = 'scm:git@github.com:cultureamp/kestrel.git'
+                    developerConnection = 'scm:git@github.com:cultureamp/kestrel.git'
+                    url = 'https://github.com/cultureamp/kestrel'
+                }
+
+                licenses {
+                    license {
+                        name 'The Apache License, Version 2.0'
+                        url 'http://www.apache.org/licenses/LICENSE-2.0.txt'
```

Comment:
Should you check in a `LICENSE` file for the GitHub repo

Reply:
Ah yeah I've seen that in other repo's - what's the reason for that over linking to the file this way?

Reply:
When you check in a LICENSE file it shows up in the about section on the Github repo. This is useful for someone who might want to fork the repository. GitHub also displays a nice overview of the license on the repository - https://github.com/ktorio/ktor/blob/master/LICENSE